packages/frontend/model/extract-ga.ts

```diff
@@ -1,43 +1,52 @@
-// export interface GADataType {
-//     pillar: string;
-//     webTitle: string;
-//     section: string;
-//     contentType: string;
-//     commissioningDesks: string;
-//     contentId: string;
-//     authorIds: string;
-//     keywordIds: string;
-//     toneIds: string;
-//     seriesId: string;
-//     isHosted: string;
-//     edition: string;
-//     beaconUrl: string;
-// }
+// All GA fields should  fall back to default values -
+import { findPillar } from './find-pillar';
 
-// // All GA fields should  fall back to default values -
-// import { getString, getBoolean } from './validators';
-// import { findPillar } from './find-pillar';
+const getMultipleTags = (
+    tags: CAPIType['tags'],
+    tagTypeIs: 'Contributor' | 'Keyword' | 'Tone' | 'Series', // Lets make a decision to keep this tag getter small and well defined, we don't really want to use tags
+): TagType['id'] | '' => {
+    const tagArr = tags.filter(tag => tag.type === tagTypeIs);
+    const arrOfvalues =
+        tagArr.length > 0 &&
+        tagArr.reduce(
+            (prev: Array<TagType['id']>, tag) => prev.concat(tag.id),
+            [],
+        );
 
-// // we should not bring down the website if a trackable field is missing!
-// export const extract = (data: {}): GADataType => {
-//     const edition = getString(data, 'page.edition', '').toLowerCase();
+    return (arrOfvalues && arrOfvalues.join(',')) || '';
+};
 
-//     return {
-//         webTitle: getString(data, 'page.webTitle', ''),
-//         pillar: findPillar(getString(data, 'page.pillar', '')) || 'news',
-//         section: getString(data, 'page.section', ''),
-//         contentType: getString(data, 'page.contentType', '')
-//             .toLowerCase()
-//             .split(' ')
-//             .join(''),
-//         commissioningDesks: getString(data, 'page.commissioningDesks', ''),
-//         contentId: getString(data, 'page.contentId', ''),
-//         authorIds: getString(data, 'page.authorIds', ''),
-//         keywordIds: getString(data, 'page.keywordIds', ''),
-//         toneIds: getString(data, 'page.toneIds', ''),
-//         seriesId: getString(data, 'page.seriesId', ''),
-//         isHosted: getBoolean(data, 'page.meta.isHosted', false).toString(),
-//         edition: edition === 'int' ? 'international' : edition,
-//         beaconUrl: getString(data, 'site.beaconUrl', ''),
-//     };
-// };
+// Annoyingly we ping GA with commissioningdesk as the title of the tag, not the id so handle that seprate
+const getCommissioningDesk = (
+    tags: CAPIType['tags'],
+): TagType['title'] | '' => {
+    const tag = tags.find(thisTag =>
+        thisTag.id.includes('tracking/commissioningdesk'),
+    );
+    return (tag && tag.title) || '';
+};
+
+// we should not bring down the website if a trackable field is missing!
+export const extract = (data: CAPIType): GADataType => {
```

Comment:
Does it make sense to move all this to Frontend at some point (maybe even share the model?) - If so, I can create a ticket...

Reply:
So hmm, I would say that tracking stuff is part of the rendering layer / derived from the model along with config, but it is blurry I agree.

Reply:
Yeah, though I think a reasonable step is to move more of the fields *used* in the tracking to the model - i.e remove use of tags - which means we remove the need for the logic but can still derive the analytics from the model in the rendering layer.